pkg/networking/subnet_resolver.go

```diff
@@ -364,6 +373,9 @@ func (r *defaultSubnetsResolver) validateSubnetsMinimalCount(subnets []*ec2sdk.S
 // computeSubnetsMinimalCount returns the minimal count requirement for subnets.
 func (r *defaultSubnetsResolver) computeSubnetsMinimalCount(subnetLocale subnetLocaleType, resolveOpts SubnetsResolveOptions) int {
 	minimalCount := 1
+	if resolveOpts.ALBSingleSubnet {
```

Comment:
Maybe add a check that the LBType is `elbv2model.LoadBalancerTypeApplication`

Reply:
I am not sure whether is necessary to add the check of elbv2model.LoadBalancerTypeApplication here since if the type is NLB the minimalCount will be returned with 1 anyway, but it might make this code change more readable?